### ISSUE TITLE
Error: TimeZoneNotFound_MissingData, GMT+0200

### ISSUE THREAD

**Issue by geertgeerits** [jfversluis/Plugin.Maui.CalendarStore#40]
If there is one calendar with a missing time zone (or some other error), event retrieval will be interrupted with an error message.
Is it possible to skip that event and still continue retrieving the other events?
See the error report in attachment.
Thank you in advance.
[report-2517058972789409999-a8de89ec-8cbe-44ad-b14e-0fa7246ffde9.json.txt](https://github.com/jfversluis/Plugin.Maui.CalendarStore/files/12802587/report-2517058972789409999-a8de89ec-8cbe-44ad-b14e-0fa7246ffde9.json.txt)

**Comment by jfversluis** [comment]
Could you provide a bit more information here? What is the code that triggers this?

Is this an event without time zone information or a calendar?

**Comment by geertgeerits** [comment]
It is an event in a particular calendar where that error occurred. Apparently there is an event in that calendar with an error in it or without a time zone.
The error occurs in the following code:
events = await CalendarStore.Default.GetEvents(startDate: dtpDateStart.Date, endDate: dtpDateEnd.Date.AddDays(1));

**Comment by jfversluis** [comment]
Interesting, because looking at the code whenever there is no time zone information it should default to a default time zone.

It's a bit hard to guess what might be going wrong though... Would you be by any chance be able to debug it with the source from his project? I'd be very curious what is going on around this area: https://github.com/jfversluis/Plugin.Maui.CalendarStore/blob/main/src/Plugin.Maui.CalendarStore/CalendarStore.macios.cs#L415

Namely what the value for `timezone` is.

**Comment by geertgeerits** [comment]
That will be difficult. This happened on my daughter's iPhone where I tested the app via TestFlight. There was a calendar that she herself had no knowledge of that was called 'Freetime'. That calendar has now been removed.

**Comment by jfversluis** [comment]
Yeah that's the interesting thing. Where did that calendar come from and what were the settings! Without it, it's pretty much guessing what the fix should be here...

Let me know if you do get any more information on this!

**Comment by jfversluis** [comment]
Since there haven't been a lot of other reports about this closing this for now. Will revisit if there is someone that can provide more information.
